// redux/src/index.ts
// functions
import { createStore, legacy_createStore } from 'redux/src/createStore'
import combineReducers from 'redux/src/combineReducers'
import bindActionCreators from 'redux/src/bindActionCreators'
import applyMiddleware from 'redux/src/applyMiddleware'
import compose from 'redux/src/compose'
import isAction from 'redux/src/utils/isAction'
import isPlainObject from 'redux/src/utils/isPlainObject'
import __DO_NOT_USE__ActionTypes from 'redux/src/utils/actionTypes'

// types
// store
export {
  Dispatch,
  Unsubscribe,
  Observable,
  Observer,
  Store,
  StoreCreator,
  StoreEnhancer,
  StoreEnhancerStoreCreator
} from 'redux/src/types/store'
// reducers
export {
  Reducer,
  ReducersMapObject,
  StateFromReducersMapObject,
  ReducerFromReducersMapObject,
  ActionFromReducer,
  ActionFromReducersMapObject,
  PreloadedStateShapeFromReducersMapObject
} from 'redux/src/types/reducers'
// action creators
export { ActionCreator, ActionCreatorsMapObject } from 'redux/src/types/actions'
// middleware
export { MiddlewareAPI, Middleware } from 'redux/src/types/middleware'
// actions
export { Action, UnknownAction, AnyAction } from 'redux/src/types/actions'

export {
  createStore,
  legacy_createStore,
  combineReducers,
  bindActionCreators,
  applyMiddleware,
  compose,
  isAction,
  isPlainObject,
  __DO_NOT_USE__ActionTypes
}
